contracts/modules/Checkpoint/WeightedVoteCheckpoint.sol

```diff
@@ -0,0 +1,148 @@
+        require(ballots[_ballotId].isActive == true, "This ballot is deactiveated.");
+        uint256 checkpointId = ballots[_ballotId].checkpointId;
+        uint256 weight = ISecurityToken(securityToken).balanceOfAt(msg.sender,checkpointId);
+        require(weight > 0, "Token Holder balance is zero.");
+        ballots[_ballotId].voteByAddress[msg.sender].time = now;
+        ballots[_ballotId].voteByAddress[msg.sender].weight = weight;
+        ballots[_ballotId].voteByAddress[msg.sender].vote = _vote;
+        ballots[_ballotId].numVotes = ballots[_ballotId].numVotes.add(1);
+        if (_vote == true) {
+            ballots[_ballotId].cumulativeYes = ballots[_ballotId].cumulativeYes.add(weight);
+        } else {
+            ballots[_ballotId].cumulativeNo = ballots[_ballotId].cumulativeNo.add(weight);
+        }
+        emit VoteCasted(_ballotId, now, msg.sender, weight, _vote);
+        return true;
+    }
+
+    /**
+     * @notice Allows the token issuer to create a ballot
+     * @param _duration The duration of the voting period in seconds
+     * @return bool success
+     */
+    function createBallot(uint256 _duration) public onlyOwner returns (bool) {
+        uint256 ballotId = ballots.length;
+        uint256 checkpointId = ISecurityToken(securityToken).createCheckpoint();
+        uint256 currentSupply = ISecurityToken(securityToken).totalSupply();
+        uint256 endTime = now.add(_duration);
+        ballots.push(Ballot(checkpointId,currentSupply,now,endTime,0,0,0,true));
+        emit BallotCreated(now, endTime, ballotId, checkpointId);
+        return true;
+    }
+
+    /**
+     * @notice Allows the token issuer to create a ballot with custom settings
+     * @param _startTime Start time of the voting period in Unix Epoch time
+     * @param _endTime End time of the voting period in Unix Epoch time
+     * @param _checkpointId Index of the checkpoint to use for token balances
+     * @return bool success
+     */
+    function createCustomBallot(uint256 _startTime, uint256 _endTime, uint256 _checkpointId) public onlyOwner returns (bool) {
+        require(_endTime >= _startTime);
+        uint256 ballotId = ballots.length;
+        uint256 supplyAtCheckpoint = ISecurityToken(securityToken).totalSupplyAt(_checkpointId);
+        ballots.push(Ballot(_checkpointId,supplyAtCheckpoint,_startTime,_endTime,0,0,0, true));
+        emit BallotCreated(_startTime, _endTime, ballotId, _checkpointId);
+        return true;
+    }
+
+    /**
+     * @notice Allows the token issuer to set the active stats of a ballot
+     * @param _ballotId The index of the target ballot
+     * @param _isActive The bool value of the active stats of the ballot
+     * @return bool success
+     */
+    function setActiveStatsBallot(uint256 _ballotId, bool _isActive) public onlyOwner returns (bool) {
+        require(now > ballots[_ballotId].startTime && now < ballots[_ballotId].endTime, "Voting period is not active.");
```

Comment:
I think we should allow this function to be called before voting period has started.